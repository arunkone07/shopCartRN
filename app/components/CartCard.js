import React from "react";

import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";

function CartCard({ title, price, image, counter, handleIncrement, handleDecrement, handleDelete }) {

  return (
    <View style={styles.card}>

      <Image
        resizeMode="contain"
        style={styles.image}
        source={{ uri: image }}
      />

      <View style={styles.detailsContainer}>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.price}>{price}</Text>


        <TouchableOpacity
          testID="delete-button"
          style={styles.delete}
          onPress={handleDelete}
        >
          <Text>D</Text>
        </TouchableOpacity>


        <TouchableOpacity
          testID="increment-button"
          style={styles.increment}
          onPress={handleIncrement}
        >
          <Text>+</Text>
        </TouchableOpacity>

        <Text style={styles.counter}>{counter}</Text>

        {counter > 0 && (
          <TouchableOpacity
            testID="decrement-button"
            style={styles.decrement}
            onPress={handleDecrement}
          >
            <Text>-</Text>
          </TouchableOpacity>
        )}

      </View>

    </View>
  );
}


const styles = StyleSheet.create({
  card: {
    marginBottom: 20,
    borderRadius: 15,
    backgroundColor: "white",
    overflow: "hidden",
  },
  counter: {
    alignSelf: "flex-end",
    position: "absolute",
    bottom: 15,
    right: 65,
    backgroundColor: "white",
    paddingVertical: 15,
    paddingHorizontal: 20,
  },
  decrement: {
    alignSelf: "flex-end",
    position: "absolute",
    bottom: 15,
    right: 110,
    backgroundColor: "orange",
    paddingVertical: 15,
    paddingHorizontal: 20,
  },
  delete: {
    alignSelf: "flex-end",
    position: "absolute",
    bottom: 15,
    right: 150,
    backgroundColor: "red",
    paddingVertical: 15,
    paddingHorizontal: 20,
  },
  detailsContainer: {
    padding: 20,
  },
  image: {
    width: "100%",
    height: 200,
  },
  increment: {
    alignSelf: "flex-end",
    position: "absolute",
    bottom: 15,
    right: 15,
    backgroundColor: "orange",
    paddingVertical: 15,
    paddingHorizontal: 20,
  },
  price: {
    color: "red",
    fontWeight: "bold",
  },
  title: {
    marginBottom: 7,
  },
});


export default CartCard;