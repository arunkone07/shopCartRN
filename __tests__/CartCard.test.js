import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import CartCard from '../app/components/CartCard';

describe('CartCard-1 test', () => {
  test('should match snapshot', () => {
    const title = 'Product 1';
    const price = '10';
    const image = 'https://shop.teamsg.in/wp-content/uploads/2021/04/Blazetech-yellow-small-2-4-scaled.jpg';
    const counter = 1;

    let card = render(
    <CartCard
      title={title} 
      price={price} 
      image={image} 
      counter={counter} 
      onIncrement={()=>{}}
      onDecrement={()=>{}}
    />
    );

    expect(card.toJSON()).toMatchSnapshot();
  });

  test('Buttons test', () => {
    const {getByTestId} = render(<CartCard/>);

    const incrementButton = getByTestId('increment-button');
    // const decrementButton = getByTestId('decrement-button');
    const deleteButton = getByTestId('delete-button');
    expect(incrementButton  && deleteButton).toBeTruthy();
  });

  test('Cart events test', () => {
    const mockFn = jest.fn()
    const {getByTestId} = render(
      <CartCard 
        handleIncrement={mockFn} 
        handleDecrement={mockFn}
        handleDelete={mockFn} 
        counter={1}
      />
    );

    fireEvent.press(getByTestId('increment-button'))
    fireEvent.press(getByTestId('decrement-button'))
    fireEvent.press(getByTestId('delete-button'))
    expect(mockFn).toHaveBeenCalledTimes(3)
  })
});
