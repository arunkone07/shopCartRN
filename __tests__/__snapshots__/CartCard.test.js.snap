// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartCard-1 test should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 15,
      "marginBottom": 20,
      "overflow": "hidden",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "https://shop.teamsg.in/wp-content/uploads/2021/04/Blazetech-yellow-small-2-4-scaled.jpg",
      }
    }
    style={
      {
        "height": 200,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "marginBottom": 7,
        }
      }
    >
      Product 1
    </Text>
    <Text
      style={
        {
          "color": "red",
          "fontWeight": "bold",
        }
      }
    >
      10
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "flex-end",
          "backgroundColor": "red",
          "bottom": 15,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 15,
          "position": "absolute",
          "right": 150,
        }
      }
      testID="delete-button"
    >
      <Text>
        D
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "flex-end",
          "backgroundColor": "orange",
          "bottom": 15,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 15,
          "position": "absolute",
          "right": 15,
        }
      }
      testID="increment-button"
    >
      <Text>
        +
      </Text>
    </View>
    <Text
      style={
        {
          "alignSelf": "flex-end",
          "backgroundColor": "white",
          "bottom": 15,
          "paddingHorizontal": 20,
          "paddingVertical": 15,
          "position": "absolute",
          "right": 65,
        }
      }
    >
      1
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "flex-end",
          "backgroundColor": "orange",
          "bottom": 15,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 15,
          "position": "absolute",
          "right": 110,
        }
      }
      testID="decrement-button"
    >
      <Text>
        -
      </Text>
    </View>
  </View>
</View>
`;
